CMakeLists.txt

```diff
@@ -47,16 +50,25 @@ if(${CUDA_VERSION} VERSION_GREATER_EQUAL "10.0")
 else()
   option(BUILD_NVTX "Build with NVTX profiling enabled" OFF)
 endif()
-option(BUILD_PYTHON "Build Python bindings" ON)
+option(BUILD_PYTHON "Build Python bindings" DEPS_ENABLED)
 option(BUILD_LMDB "Build LMDB readers" OFF)
-option(BUILD_JPEG_TURBO "Build with libjpeg-turbo support" ON)
-option(BUILD_LIBTIFF "Build with libtiff support" ON)
-option(BUILD_NVJPEG "Build with nvJPEG support" ON)
+option(BUILD_JPEG_TURBO "Build with libjpeg-turbo support" DEPS_ENABLED)
+option(BUILD_LIBTIFF "Build with libtiff support" DEPS_ENABLED)
+option(BUILD_NVJPEG "Build with nvJPEG support" DEPS_ENABLED)
 option(BUILD_NVOF "Build with NVIDIA OPTICAL FLOW SDK support" ON)
-option(BUILD_NVDEC "Build with NVIDIA NVDEC support" ON)
-option(BUILD_LIBSND "Build with suport for libsnd library" ON)
+option(BUILD_NVDEC "Build with NVIDIA NVDEC support" DEPS_ENABLED)  # Disabling for nodeps (requires ffmpeg)
+option(BUILD_LIBSND "Build with suport for libsnd library" DEPS_ENABLED)
 option(BUILD_NVML "Build with NVIDIA Management Library (NVML) support" ON)
 option(BUILD_FFTS "Build with ffts support" ON)
+option(BUILD_OPENCV "Build with suport for OpenCV library" DEPS_ENABLED)
+option(BUILD_PROTOBUF "Build with suport for Protobuf library" DEPS_ENABLED)
+option(BUILD_DALI_CORE "Build dali_core library" ON)
+option(BUILD_DALI_KERNELS "Build dali_kernels library" ON)
```

Comment:
How about using CMakeDependentOption, as  you cannot build Pipeline and Operators without the necessary prerequisites.

Reply:
done